rpcserver.go

```diff
@@ -6777,7 +6779,48 @@ func (r *rpcServer) ForwardingHistory(ctx context.Context,
 	}
 	timeSlice, err := r.server.miscDB.ForwardingLog().Query(eventQuery)
 	if err != nil {
-		return nil, fmt.Errorf("unable to query forwarding log: %v", err)
+		return nil, fmt.Errorf("unable to query forwarding log: %v",
+			err)
+	}
+
+	// chanToPeerAlias caches previously looked up channel information.
+	chanToPeerAlias := make(map[lnwire.ShortChannelID]string)
+
+	// Helper function to extract a peer's node alias given its SCID.
+	getRemoteAlias := func(chanID lnwire.ShortChannelID) string {
+		// If we'd previously seen this chanID then return the cached
+		// peer alias.
+		if peerAlias, ok := chanToPeerAlias[chanID]; ok {
+			return peerAlias
+		}
+
+		// Else call the server to look up the peer alias.
+		edge, err := r.GetChanInfo(ctx, &lnrpc.ChanInfoRequest{
+			ChanId: chanID.ToUint64(),
+		})
+		if err != nil {
+			return ""
+		}
+
+		remotePub := edge.Node1Pub
+		if r.selfNode.String() == edge.Node1Pub {
+			remotePub = edge.Node2Pub
+		}
+
+		vertex, err := route.NewVertexFromStr(remotePub)
+		if err != nil {
+			return ""
```

Comment:
quick question: why not just return an error from this func so that we can return it to the user if something goes wrong?

Reply:
Since the helper is called for a range of events I didn't want to interrupt the alias enrichment if a single lookup fails. I am also not sure how to communicate multiple lookup errors in a range of events. I am unsure if this would be worth logging though?

Reply:
@ellemouton I've added an error return value to the helper function which could be evaluated by the caller. In case of an error is there any particular behavior you would suggest. Should we log errors returned from the helper?

Reply:
I think if it results in an error, then return that error.

Reply:
Just want to avoid a strange return result that has some populated aliases and some empty ones. 

Returning the error will help with debugging any issues.

If there is an error returned, the user can always just default to calling the command without the new option

Reply:
Ok got it. So the error *is* now returned but not evaluated or logged. Should the error message be printed in place of the peer alias or should we log the error?

Reply:
what I meant is that it should be evaluated and returned from ForwardingHistory

Reply:
Thanks for the clarification @ellemouton. I've changed it so that `FowardingHistory` now returns the error.